pkg/extender/extender.go

```diff
@@ -696,8 +698,14 @@ func (e *Extender) processPrioritizeRequest(w http.ResponseWriter, req *http.Req
 				storklog.PodLog(pod).Debugf("Volume %v allocated in regions: %v", volume.VolumeName, regionInfo.PreferredLocality)
 
 				for k8sNodeIndex, node := range args.Nodes.Items {
-					storageNode := k8sNodeIndexStorageNodeMap[k8sNodeIndex]
-					priorityMap[node.Name] += int(e.getNodeScore(node, volume, &rackInfo, &zoneInfo, &regionInfo, storageNode))
+					if storageNode, keyExists := k8sNodeIndexStorageNodeMap[k8sNodeIndex]; !keyExists {
+						// the k8sNodeIndex's corresponding node isn't available to the driver,
+						// so we shouldn't schedule pods on this node
+						// we will assign a negative score here and set it to 0 when encoding the response
+						priorityMap[node.Name] = -1
+					} else {
+						priorityMap[node.Name] += int(e.getNodeScore(node, volume, &rackInfo, &zoneInfo, &regionInfo, storageNode))
+					}
```

Comment:
nit.. This does work. However I think it would have been cleaner if the "if" and "else" were swapped.
 What I mean is this
```
if storageNode, keyExists := k8sNodeIndexStorageNodeMap[k8sNodeIndex]; keyExists {
  priorityMap[node.Name] += int(e.getNodeScore(node, volume, &rackInfo, &zoneInfo, &regionInfo, storageNode))
}
else{
  priorityMap[node.Name] = -1
}
```

Reply:
Sure will address